infuriated-mink/Semester4-Final-Backend
# Set up EC2 instance and deploy Dockerized Spring Boot application

## Description

This pull request includes the following changes:

- Set up an EC2 instance on AWS using Amazon Linux 2023 AMI.
- Configured security group rules to allow inbound traffic on ports 22 (SSH), 80 (HTTP), and 8080 (Custom TCP).
- Installed Docker on the EC2 instance and started the Docker service.
- Pulled the Docker image for the Spring Boot application from Docker Hub.
- Ran the Docker container with environment variables to connect to the RDS instance.
- Verified that the application is running and accessible via HTTP on port 8080.

## Changes
- **Dockerfile**: Configuration for Docker image.

## Verification

- Verified that the Spring Boot application is running and accessible via HTTP on port 8080.
- Tested API endpoints using the provided URL.
- Verified that the React application successfully fetches and displays data from the backend.

## Screenshots

<img width="1407" alt="Screenshot 2024-08-02 at 4 46 09 PM" src="https://github.com/user-attachments/assets/f61fba93-741d-4e95-a2aa-c33d9c3646f7">


## How to Test

1. SSH into the EC2 instance.
2. Verify that Docker is installed and running.
3. Pull the Docker image from Docker Hub.
4. Run the Docker container with the necessary environment variables.
5. Access the application using the provided URL: `http://ec2-3-149-229-73.us-east-2.compute.amazonaws.com:8080`.
6. Verify that the application is running and accessible.
7. Test the React application to ensure it fetches and displays data from the backend.

## Additional Notes

- Ensure that the security group rules are correctly configured to allow inbound traffic on port 8080.
- Verify that the Network ACLs allow inbound and outbound traffic on port 8080.
- Use HTTP to access the application, not HTTPS.
